static/bonescript.js

```diff
@@ -3,28 +3,36 @@ function require(file) {
 }
 
 function setTargetAddress(address, handlers) {
+    scriptLoaded = false;
     var url = address;
-    url = url.replace(/^(http:\/\/|https:\/\/)*/, 'http://');
+    url = url.replace(/^(http:\/\/|https:\/\/)*/, 'https://');
     url = url.replace(/(\/)*$/, '/bonescript.js');
-    loadScript(url, addHandlers);
-    function loadScript(url, onload) {
-        try {
-            var head = document.getElementsByTagName('head')[0];
-            var script = document.createElement('script');
-            script.type = 'text/javascript';
-            script.src = url;
-            script.charset = 'UTF-8';
-            var scriptObj = head.appendChild(script);
-            scriptObj.addEventListener('error', onerror);
-            scriptObj.onload = onload;
-        } catch(ex) {
-            onerror(ex);
-        }
-
-        function onerror(err) {
-            //console.log('Unable to load: ' + url);
-        }
+    
+    loadScript(url);
+    if(scriptLoaded==false)
```

Comment:
I don't think this could work. loadScript() works asynchronously. this condition would always be true (scriptLoaded would always be false). Not sure how you tested to understand your new logic worked.